app/session.js

```diff
@@ -76,46 +76,46 @@ module.exports = class Session extends EventEmitter {
   }
 
   getTitle() {
-    if (process.platform === 'win32') {
-      return;
-    }
-
-    if (this.fetching) {
-      return;
-    }
-    this.fetching = true;
-
-    let tty = this.pty.stdout.ttyname;
-    tty = tty.replace(/^\/dev\/tty/, '');
-
-    // try to exclude grep from the results
-    // by grepping for `[s]001` instead of `s001`
-    tty = `[${tty[0]}]${tty.substr(1)}`;
-
-    // TODO: limit the concurrency of how many processes we run?
-    // TODO: only tested on mac
-    exec(`ps uxac | grep ${tty} | head -n 1`, (err, out) => {
-      this.fetching = false;
-      if (this.ended) {
-        return;
-      }
-      if (err) {
-        return;
-      }
-      let title = out.split(' ').pop();
-      if (title) {
-        title = title.replace(/^\(/, '');
-        title = title.replace(/\)?\n$/, '');
-        if (title !== this.lastTitle) {
-          this.emit('title', title);
-          this.lastTitle = title;
-        }
-      }
-
-      if (this.subscribed) {
-        this.titlePoll = setTimeout(() => this.getTitle(), TITLE_POLL_INTERVAL);
-      }
-    });
```

Comment:
Sure, I didn't mean to leave this part commented... but I think it could be useful to replace this code with some logic to set the window title in a way that it reflect the title of the current tab.e.g. iTerm works this way...